### ISSUE TITLE
Add feature to change the History

### ISSUE THREAD

**Issue by ramack** [ramack/ActivityDiary#28]
Allow inserting activities in the history view.

**Comment by Echsecutor** [comment]
> change start and end times (with or without adjacent diary entries)

This would be a very helpful feature! Often I realized that I changed my activity like half an hour ago without pushing the button.

**Comment by ramack** [comment]
Do you have an idea how this would beat be implemented? E.g how should we get to the possibility to change the time? - this issue was meant to be in the diary/history view, but the usecase you mention should probably be easier to access that...

Suggestions welcome!

**Comment by ramack** [comment]
Thanks BTW for the feedback. This is very helpful!

**Comment by ramack** [comment]
I am not an expert in UI design. Is someone willing to suggest how the history edit can be implemented? - a simple picture could help already a lot.

For the current activity we could add a button, but another Floating action button will probably be an overload and just b tapping on the duration is not visible enough... on the other side decreasing the 
Selection view for the next activity is also not so good.

**Comment by ramack** [comment]
ok, my idea would be to
in the history/diary:
- left swipe on an activity reveals a trash and deletes the entry on finalizing the swipe (if it is not reversed before the finger lifted) the history shows an "unused slot" with "<no activity>"; right swipe the same
- two finger swipe will insert an activity between the two entries which are closest to the center of the two fingers. Duration is 5min (but configurable in preferences), cutting half of it from the pre- and half from the successor.
- the action bar in the diary gets an "edit" icon, on selecting it the screen is "filled" with floating action buttons:
  * a multi-action FAB is shown on the left side (...) pressing it will make it multiple possibilities: delete activity - trash, edit Note, add picture
  * one with up and down arrows between all the entries (in the screen center), on dragging them up and down the edge between the two adjacent entries is shifted (affecting start time of the later one and end time of the earlier entry), during the drag the duration of both activities and the time (which is new end and start) is displayed in a toast at the finger position, if one of the activities is crossing a date boundary not only time, but also weekday and date is shown.
  * during editing the height of the entries is scaled with their length (short entries are small, long are very height)
  * on the right edge of the screen an "insertion arrow - with a +" is shown at each boundary between entries

In the "edit mode" there are a lot of floating action buttons, but if they are not made too prominent it might work out I think.

Any comments?

**Comment by ramack** [comment]
mh, editing the times graphically is hard to get right and intuitive. So I decided to go for a edit activity for now, to simplify note editing and allow to change start and end times.

Still we need further improvements, but I think they can wait a release.

**Comment by ramack** [comment]
@Echsecutor and others: what do you think about the current possibilities for editing the history? - I have some improvements in mind, but what is the top priority in your eyes?

**Comment by Echsecutor** [comment]
@ramack I have just tried 1.2.5 and I like the possibility to change durations. Thanks for adding it! My next most wanted feature would be `remove activities`. If I accidentally clicked something... ;)

**Comment by ramack** [comment]
😃 until deletion and insertion is implemented there is only one workaround: if you directly recognize that you touched the wrong activity you can use the UNDO in the snackbar after selecting the wrong one.

The real solution would be to provide a pull request, or wait.

But also highlighting it again and again is welcome, because it shows me what the users really want.

**Comment by Echolon** [comment]
I definitely suggest to inplement the remove possibility. Unless, the pretty much unusable.

**Comment by michaelsam94** [comment]
I have question in delete history entry now the deleted activities shows when search so how I get rid of specific activity completely.

**Comment by ramack** [comment]
@michaelsam94 this is intended behavior, as the activities are the list of things you can select, and deleting one will not allow to select it in future, but in the diary I want it to stay as this is documenting 'history'. What you seems to be looking for is to delete all instances of a certain activity in the diary. This has to be possible, but it is not yet implemented.

**Comment by michaelsam94** [comment]
willing to implement it?

**Comment by ramack** [comment]
Yes clearly. It is on top of the list, but I don't have much for free software these days. 

The good thing is, that anyone is free to implement it - I promise to merge a pull request!
